packages/kotti-ui/source/kotti-table/components/TableHeader.vue

```diff
@@ -51,146 +47,103 @@
 </template>
 
 <script lang="ts">
-/* eslint-disable @typescript-eslint/no-explicit-any */
-
-import { KT_TABLE, KT_STORE, KT_LAYOUT } from '../constants'
+import { KT_TABLE, KT_STORE } from '../constants'
 import { KottiTable } from '../types'
+import { computed, defineComponent, inject, ref } from 'vue'
 
 import { TableHeaderCell } from './TableHeaderCell'
 
-// eslint-disable-next-line @typescript-eslint/naming-convention
-const KtTableHeader: any = {
+export default defineComponent({
 	name: 'KtTableHeader',
 	components: { TableHeaderCell },
-	inject: { KT_TABLE, KT_STORE, KT_LAYOUT },
-	data() {
+	setup() {
+		const tableState = inject(KT_TABLE)
+		const tableStore = inject(KT_STORE)
+
+		if (!tableState || !tableStore)
+			throw new Error(
+				'TableRowCell: Component was used without providing the right contexts',
+			)
+		const columnDragOver = ref<KottiTable.Column.PropsInternal | null>(null)
+		const draggingColumn = ref<KottiTable.Column.PropsInternal | null>(null)
+
+		const isDraggable = computed(() => tableState.useColumnDragToOrder)
+		const isDraggedOver = (column: unknown) => {
+			return column === columnDragOver.value && draggingColumn.value !== column
+		}
+		const useQuickSortControl = computed(() => tableState.useQuickSortControl)
+
+		const canSort = (column: unknown) => tableStore.get('canSort', column)
+		const isSorted = (column: unknown) => tableStore.get('isSorted', column)
+
 		return {
-			draggingColumn: null,
-			columnDragOver: null,
+			canSort,
+			dragEnd: () => {
+				columnDragOver.value = null
+			},
+			dragEnter: (_event: unknown, column: KottiTable.Column.PropsInternal) => {
+				columnDragOver.value = column
+			},
+			dragStart: (_event: unknown, column: KottiTable.Column.PropsInternal) => {
+				draggingColumn.value = column
+			},
+			drop: (_event: unknown, column: KottiTable.Column.PropsInternal) => {
+				if (draggingColumn.value) {
+					tableStore.commit('orderBefore', draggingColumn.value, column)
+					draggingColumn.value = null
+				}
+				columnDragOver.value = null
+			},
+			getThClasses: (column: KottiTable.Column.PropsInternal) => {
+				return [
+					{
+						['drag-over']: isDraggedOver(column),
```

Comment:
I might be wrong but shouldn’t this be enough?

```suggestion
						'drag-over': isDraggedOver(column),
```